core/node/state_keeper/src/testonly/test_batch_executor.rs

```diff
@@ -833,3 +837,18 @@ impl BatchExecutor for MockBatchExecutor {
         Some(BatchExecutorHandle::from_raw(handle, send))
     }
 }
+
+#[derive(Debug)]
+pub struct MockReadStorageFactory;
+
+#[async_trait]
+impl ReadStorageFactory for MockReadStorageFactory {
+    async fn access_storage(
+        &self,
+        _stop_receiver: &Receiver<bool>,
+        _l1_batch_number: L1BatchNumber,
+    ) -> anyhow::Result<Option<PgOrRocksdbStorage<'_>>> {
+        // Presume that the storage is never accessed so doesn't matter what we return here
```

Comment:
Nit: If the storage is indeed never accessed, it makes sense to use `unimplemented!()` here.